examples/simple-app/src/main.ts

```diff
@@ -125,30 +126,97 @@ client.on("peerJoined", (peer) => {
   remotePeers.appendChild(clone);
   toastInfo(`New peer joined`);
 });
+
 client.on("peerUpdated", (_peer) => {});
+
 client.on("peerLeft", (peer) => {
   const peerComponent = document.querySelector(`div[data-peer-id="${peer.id}"`)!;
   peerComponent.remove();
   toastInfo(`Peer left`);
 });
+
+const setUpSimulcastCheckbox = (element: HTMLDivElement, trackId: string, encoding: "l" | "m" | "h") => {
+  const simulcastInputL: HTMLInputElement | null = element.querySelector<HTMLInputElement>(
+    `.simulcast-input-radio-${encoding}`,
+  );
+  if (!simulcastInputL) return;
+
+  simulcastInputL.setAttribute("name", `${trackId}-simulcast`);
+
+  simulcastInputL.addEventListener("click", () => {
+    if (client.getRemoteTracks()[trackId]?.simulcastConfig?.enabled) {
+      client.setTargetTrackEncoding(trackId, encoding);
+    } else {
+      console.warn("You cannot set 'targetTrackEncoding' on a track that doesn't have an active simulcast.");
+      toastInfo("You cannot set 'targetTrackEncoding' on a track that doesn't have an active simulcast.");
+    }
+  });
+};
+
+const ENCODINGS: TrackEncoding[] = ["l", "m", "h"];
+
 client.on("trackReady", (ctx) => {
   console.log("On track ready");
+  console.log({ name: "trackReady", ctx });
+  if (!ctx.trackId) return;
+
   const peerId = ctx.endpoint.id;
   const peerComponent = document.querySelector(`div[data-peer-id="${peerId}"`)!;
-  const videoPlayer: HTMLVideoElement = peerComponent.querySelector(".remote-peer-template-video")!;
+
+  const videoPlayerTemplate = document.querySelector("#remote-peer-template-video")!;
+
+  // @ts-ignore
+  const videoWrapper = videoPlayerTemplate.content.cloneNode(true);
```

Comment:
```suggestion
  const videoPlayerTemplate = document.querySelector<HTMLTemplateElement>("#remote-peer-template-video")!;

  const videoWrapper = <DocumentFragment>videoPlayerTemplate.content.cloneNode(true);
```

All subsequent `querySelector` calls will require non-null assertions. I think that it's still worth it, as the `videoWrapper` is effectively typed as `any` right now.